### PR TITLE
change switch icon to mdiToggleSwitch

### PR DESCRIPTION
**Pull request by Mariusthvdb**
from mdiFlash to mdiToggleSwitch.
reasoning: flash is not really switch related, and used on power entities, which is very suitable. ToggleSwitch speaks for itself, and could be even set in the device_class switch to be used alongside mdiToggleSwitchOff.

Outline variants are used for input_boolean, so don't interfere either

__update__

made the icons state dependent

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by GirzzlyAK** [comment]
+1

File: src/common/entity/domain_icon.ts [hunk 1]
```diff
@@ -63,6 +65,9 @@ export const domainIcon = (
     case "media_player":
       return compareState === "playing" ? mdiCastConnected : mdiCast;
 
+    case "switch":
+      return compareState === "on" ? mdiToggleSwitch : mdiToggleSwitchOff;
```
Comment (ludeeus) [inline src/common/entity/domain_icon.ts:69]: I don't think we should use this for all switches, for those with device_class `toggle` this makes sense.
For `outlet`? maybe `mdiPowerPlug` and `mdiPowerPlugOff`
For switches with assumed state, I think the current is the best variant.

Comment (Mariusthvdb) [reply src/common/entity/domain_icon.ts:69]: tbh, Ive never understood why the flash was used at all, even using assumed state, having 2 identical flashes is very unclear in its meaning.
could you please point me to the switch device_class, I've just asked about that in Discord #frontend, as I applied it and no change in icon occurs.

agree powerful would be cool to add for those
can we add:
```
if (stateObj?.attributes.device_class === "outlet") {
         return compareState === "on" ? mdiPowerPlug : mdiPowerPlugOff;
       }
```
like that?

Comment (ludeeus) [reply src/common/entity/domain_icon.ts]: it was `switch` I meant, updated.

**Comment by Mariusthvdb** [comment]
checks wont restart, even though I added mdiFlash...

File: src/common/entity/domain_icon.ts [hunk 2]
```diff
@@ -72,6 +77,15 @@ export const domainIcon = (
     case "media_player":
       return compareState === "playing" ? mdiCastConnected : mdiCast;
 
+    case "switch":
+      if (stateObj?.attributes.device_class === "outlet") {
+        return compareState === "on" ? mdiPowerPlug : mdiPowerPlugOff;
+      }
+      if (stateObj?.attributes.device_class === "switch") {
+        return compareState === "on" ? mdiToggleSwitch : mdiToggleSwitchOff;
+      }
```
Comment (bramkragten) [inline src/common/entity/domain_icon.ts:86]: Please use a `switch` statement here

File: src/common/entity/domain_icon.ts [hunk 3]
```diff
@@ -72,6 +77,26 @@ export const domainIcon = (
...
 
+    case "switch":
+      switch (stateObj?.attributes.device_class) {
+        case "outlet":
+          switch (state) {
+            case "on":
+              return mdiPowerPlug;
+            default:
+              return mdiPowerPlugOff;
+          }
```
Comment (bramkragten) [inline src/common/entity/domain_icon.ts:88]: But don't use a switch here
```suggestion
          return state === "on" ? mdiPowerPlug : mdiPowerPlugOff;
```

File: src/common/entity/domain_icon.ts [hunk 4]
```diff
@@ -72,6 +77,26 @@ export const domainIcon = (
...
+        case "switch":
+          switch (state) {
+            case "on":
+              return mdiToggleSwitch;
+            default:
+              return mdiToggleSwitchOff;
+          }
```
Comment (bramkragten) [inline src/common/entity/domain_icon.ts:95]: Same

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**